Source/WebCore/PAL/pal/spi/mac/VisionKitCoreSPI.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2022 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#import <AppKit/AppKit.h>
```

Comment:
Will we need a separate SPI header for iOS? I don't know much about VisionKit, but I'd expect it to be similar between platforms.

Reply:
Yeah, I don't think we should try to maintain separate `VisionKitCoreSPI.h` SPI headers on macOS and iOS.

Maybe we should including some pieces of `VKDefines.h`? e.g.

```
#if PLATFORM(IOS)
#import <UIKit/UIKit.h>
typedef UIImage VKImageClass;
typedef UIResponder VKResponder;
typedef UIView VKBaseViewClass;
typedef UIWindow VKWindowClass;
#elif PLATFORM(MAC)
#import <AppKit/AppKit.h>
typedef NSImage VKImageClass;
typedef NSResponder VKResponder;
typedef NSView VKBaseViewClass;
typedef NSWindow VKWindowClass;
#endif
```

Reply:
Ah, I see now — these macOS/iOS-specific headers only include API that's specific to those platforms...

In any case, I think it would probably be best to have all the declarations available via a single header, such that there's a single `VisionKitCoreSPI.h` header to import from PAL that contains all the VKC symbols.

Reply:
Makes sense! I don't think we need the full expressivity of `VKDefines` at this point (maybe I'll discover that we do when I work on the open source iOS build…), since all we need at this time is the Mac-specific `VKCImageAnalysisOverlayView`.